ansible/roles/prometheus/tasks/config.yml

```diff
@@ -85,8 +85,45 @@
     - "{{ ansitheus_custom_config_dir }}/prometheus/{{ inventory_hostname }}/prometheus.yml"
     - "{{ ansitheus_custom_config_dir }}/prometheus/prometheus.yml"
     - "{{ role_path }}/templates/prometheus.yml.j2"
+
+- name: Shard target configurations
+  inject_sharding_config:
+    source: "{{ prometheus_config_dir }}/prometheus.yml"
+    modulus: "{{ groups['prometheus'] | length }}"
+    hash_value: "{{ groups['prometheus'].index(inventory_hostname) }}"
+  when: enable_target_sharding == 'yes'
+
+- name: Find file_sd YAML configurations
+  find:
+    paths: "{{ prometheus_config_dir }}/file_sd"
+    patterns: "*.yml,*.yaml"
+    use_regex: false
+  register: file_sd_yaml
+  when:
+    - enable_target_sharding == 'yes'
```

Comment:
should use [bool](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/bool_filter.html) filter: `enable_target_sharding | bool`. Doing string comparison can go wrong: "True", "Yes",  "yEs", ... 😸